### PR TITLE
Feature/change substate id to up count

### PR DESCRIPTION
**Pull request by talekhinezh**
* Change SubstateId to refer to UP index rather than instruction index

### CODE REVIEW

File: radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/ConstraintMachine.java [hunk 1]
```diff
@@ -120,11 +121,11 @@ public ImmutableAddrs readableAddrs() {
 			return store.loadUpParticle(dbTxn, substateId, deserialization);
 		}
 
-		public void bootUp(int instructionIndex, Substate substate, Supplier<ByteBuffer> buffer) {
-			localUpParticles.put(instructionIndex, Pair.of(substate, buffer));
+		public void bootUp(Substate substate, Supplier<ByteBuffer> buffer) {
```
Comment (michalzajda) [inline radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/ConstraintMachine.java:124]: OK, completely independent index for up outputs

File: radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/REInstruction.java [hunk 2]
```diff
@@ -90,30 +90,30 @@ public Object read(Txn txn, int index, ByteBuffer buf, SubstateDeserialization d
...
 					throw new DeserializeException("Bad local index: " + index);
 				}
-				return SubstateId.ofSubstate(txn.getId(), index);
+				return SubstateId.ofSubstate(parserState.txnId(), index);
 			}
 		},
 		DOWNALL((byte) 8, REOp.DOWNALL) {
 			@Override
-			public Object read(Txn txn, int i, ByteBuffer b, SubstateDeserialization d) throws DeserializeException {
+			public Object read(REParser.ParserState parserState, ByteBuffer b, SubstateDeserialization d) throws DeserializeException {
```
Comment (michalzajda) [inline radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/REInstruction.java:109]: a bit of inconsistency with "b" and "buf"

**Review by michalzajda — APPROVED** [review]

**Review by siy — APPROVED** [review]
LGTM

**Review by iamyulong — APPROVED** [review]
LGTM

**Review by LukasGasior1 — APPROVED** [review]
LGTM

**Pull request merged**